File: src/main/java/dev/misei/einfachml/controller/dto/PlotBoard.java
package dev.misei.einfachml.controller.dto;

import dev.misei.einfachml.repository.model.PredictedData;
import lombok.Data;

import java.util.List;

@Data
public class PlotBoard {

    List<PredictedData> lastEpochPredicted;
    List<Double> mseErrors;
}

File: src/main/resources/static/classes/PlotBoard.js
import {PredictedData} from "./PredictedData.js";

export class PlotBoard {
    constructor(lastEpochPredicted, mseErrors) {
        this.lastEpochPredicted = lastEpochPredicted;  // Array of PredictedData
        this.mseErrors = mseErrors;  // Array<number>
    }

    static fromJson(jsonObj) {
        return new PlotBoard(
            //jsonObj.lastEpochPredicted.map(item => PredictedData.fromJson(item)),
            jsonObj.mseErrors
        );
    }
}